src/lib/libcore.js

```diff
@@ -1417,11 +1417,18 @@ addToLibrary({
     }
   },
 
-  emscripten_log__deps: ['$formatString', '$emscriptenLog'],
+  emscripten_log__deps: ['$stackAlloc', 'vasprintf', '$emscriptenLog', 'free'],
   emscripten_log: (flags, format, varargs) => {
-    var result = formatString(format, varargs);
-    var str = UTF8ArrayToString(result);
-    emscriptenLog(flags, str);
+    var bufPtrPtr = stackAlloc({{{ POINTER_SIZE }}});
+    var len = _vasprintf(bufPtrPtr, format, varargs);
+    assert(len > 0);
+    var bufPtr = {{{ makeGetValue('bufPtrPtr', 0, '*') }}};
+    try {
+      var str = UTF8ToString(bufPtr, len);
+      emscriptenLog(flags, str);
+    } finally {
+      _free(bufPtr);
+    }
```

Comment:
It could be argued that converting `emscripten_log` to have more native dependencies a bad idea... but we already have  emscripten_outf / emscripten_errf / emscripten_dbgf that all depend on native formatting so maybe its fine.

A better fix here might be have `emscripten_log` implemented in native code and then only out to JS once the string is formatting.  However, IIRC emscripten_log itself is not much used at all, and the lower level emscirpten_out emscripten_err, emscripten_dbg, emscripten_console_log, and emscripten_console_err should be preferred.

Reply:
Actually, yes I think it would make sense to move this code to `system/lib/libc/emscripten_console.c` and then have the JS version take the fully formatted string.

Reply:
> A better fix here might be have `emscripten_log` implemented in native code

Yeah I thought of / started doing that first, but then it seemed it blows the scope of PR a bit too much, so reverted and figured it's better to ship just the removal of libformatString first.